### PR TITLE
Allow registering the deviceApiKey in a non-compatible OS by making the apiKey equal the deviceApiKey, and add an fsync to all config.json writes

### PR DESCRIPTION
**Pull request by pcarranzav**
Change-Type: patch
Signed-off-by: Pablo Carranza Velez <pablo@resin.io>

### CODE REVIEW

File: src/bootstrap.coffee [hunk 1]
```diff
@@ -64,11 +73,15 @@ exchangeKey = ->
 		.then (device) ->
 			if not device?
 				throw new ExchangeKeyError("Couldn't fetch device with provisioning key")
-			# We found the device, we can try to generate a working device key for it
-			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}")
+			# We found the device, we can try to register a working device key for it
+			userConfig.deviceApiKey ?= deviceRegister.generateUniqueKey()
+			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}", { json: true, body: apiKey: userConfig.deviceApiKey })
```
Comment (Page-) [inline src/bootstrap.coffee]: Can you split the options object onto new lines please, it's really easy to miss that we pass extra options atm as it's after a long string

Comment (pcarranzav) [inline src/bootstrap.coffee]: Done

**Comment by pcarranzav** [comment]
@Page- I made a few changes:
* Fixed getting the osVersion before checking if the OS has deviceApiKey support during provisioning
* Removed the `userConfig.deviceApiKey = body` when doing the key exchange, as this PR will depend on the resin-api PR anyways.

**Comment by pcarranzav** [comment]
Connects to https://github.com/resin-io/hq/pull/938/files

**Pull request merged**